pkg/crawler/crawler.go

```diff
@@ -221,6 +227,27 @@ func (c *Crawler) Visit(ctx context.Context, url string) error {
 	return nil
 }
 
+// To avoid a large number of requests to the server, we should skip already saved artifacts (if the start date is specified).
+// P.S. We do not need to check for updates, since artifacts are immutable
+// see https://central.sonatype.org/publish/requirements/immutability
```

Comment:
The document seems relevant to the sonatype repository. Is this repository also immutable?
https://mvnrepository.com/repos/central

Reply:
I didn't find official info about that.
But i saw answer about that - https://stackoverflow.com/questions/40739939/dropping-a-release-from-public-maven-central

Also indirect evidence is [this answer](https://github.com/protocolbuffers/protobuf/issues/18484#issuecomment-2417822438) (they had the same sha1 for several artifacts):
instead of changing the file - they release a new version